finality-aleph/src/data_io/data_provider.rs

```diff
@@ -307,20 +307,21 @@ struct DataProvider<B: BlockT> {
 //    at most MAX_DATA_BRANCH_LEN.
 #[async_trait]
 impl<B: BlockT> aleph_bft::DataProvider<AlephData<B>> for DataProvider<B> {
-    async fn get_data(&mut self) -> AlephData<B> {
-        let data = (*self.data_to_propose.lock().await).clone();
+    async fn get_data(&mut self) -> Option<AlephData<B>> {
+        let data_to_propose = (*self.data_to_propose.lock().await).take();
```

Comment:
In case the data_to_propose is not updated, previously multiple calls to get_data would return the same proposals (because of the clone), now only the first call will return data, and the next ones will return None.

Reply:
Just making sure it will not result in any unexpected behaviour changes in the aleph-bft

Reply:
That was intended :) There is no point in ordering the same hash multiple times.